PiT 24.4: cannot run any hilla project with 24.4.0.alpha7

### Describe the bug

When trying to migrate a hilla project from 2.5 to 24.4 I get a npm installation error

```
Command `/usr/local/bin/npm --no-update-notifier --no-audit --scripts-prepend-node-path=true --ignore-scripts install` failed:
npm ERR! code ERESOLVE
npm ERR! ERESOLVE could not resolve
npm ERR!
npm ERR! While resolving: no-name@undefined
npm ERR! Found: react-router@6.20.1
npm ERR! node_modules/react-router
npm ERR!   react-router@"6.20.1" from react-router-dom@6.20.1
npm ERR!   node_modules/react-router-dom
npm ERR!     react-router-dom@"6.22.1" from the root project
npm ERR!
npm ERR! Could not resolve dependency:
npm ERR! dev @vaadin/hilla-file-router@"24.4.0-alpha7" from the root project
npm ERR!
npm ERR! Conflicting peer dependency: react-router@6.22.2
npm ERR! node_modules/react-router
npm ERR!   peer react-router@"^6.21.1" from @vaadin/hilla-file-router@24.4.0-alpha7
npm ERR!   node_modules/@vaadin/hilla-file-router
npm ERR!     dev @vaadin/hilla-file-router@"24.4.0-alpha7" from the root project
npm ERR!
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
npm ERR!
npm ERR!
```

### Expected-behavior

Project compiles

### Reproduction

```
curl -s -f 'https://start.vaadin.com/dl?&preset=react-tutorial&projectName=react-tutorial' -o react-tutorial.zip
unzip react-tutorial.zip
cd react-tutorial
## Change Maven property hilla.version from 2.5.6 -> 24.4.0.alpha7
```

Follow instructions in https://github.com/vaadin/docs/issues/3186 to migrate the project, here the diff files:

```
diff --git a/frontend/components/placeholder/Placeholder.tsx b/frontend/components/placeholder/Placeholder.tsx
index c9443d2..286038b 100644
--- a/frontend/components/placeholder/Placeholder.tsx
+++ b/frontend/components/placeholder/Placeholder.tsx
@@ -1,4 +1,4 @@
-import { ProgressBar } from '@hilla/react-components/ProgressBar.js';
+import { ProgressBar } from '@vaadin/react-components/ProgressBar.js';

 export default function Placeholder() {
   return <ProgressBar indeterminate={true} className="m-0" />;
diff --git a/frontend/views/MainLayout.tsx b/frontend/views/MainLayout.tsx
index 161ec14..65eb768 100644
--- a/frontend/views/MainLayout.tsx
+++ b/frontend/views/MainLayout.tsx
@@ -1,5 +1,5 @@
-import { AppLayout } from '@hilla/react-components/AppLayout.js';
-import { DrawerToggle } from '@hilla/react-components/DrawerToggle.js';
+import { AppLayout } from '@vaadin/react-components/AppLayout.js';
+import { DrawerToggle } from '@vaadin/react-components/DrawerToggle.js';
 import Placeholder from 'Frontend/components/placeholder/Placeholder.js';
 import { useRouteMetadata } from 'Frontend/util/routing.js';
 import { Suspense, useEffect } from 'react';
```

```
diff --git a/pom.xml b/pom.xml
index 2e5c380..07a2d8b 100644
--- a/pom.xml
+++ b/pom.xml
@@ -11,8 +11,10 @@

     <properties>
         <java.version>17</java.version>
-        <hilla.version>2.5.6</hilla.version>
+        <vaadin.version>24.4.0.alpha7</vaadin.version>
     </properties>
+    <pluginRepositories><pluginRepository><id>v</id><url>https://maven.vaadin.com/vaadin-prereleases</url></pluginRepository></pluginRepositories>
+    <repositories><repository><id>v</id><url>https://maven.vaadin.com/vaadin-prereleases</url></repository></repositories>

     <parent>
         <groupId>org.springframework.boot</groupId>
@@ -23,9 +25,9 @@
     <dependencyManagement>
         <dependencies>
             <dependency>
-                <groupId>dev.hilla</groupId>
-                <artifactId>hilla-bom</artifactId>
-                <version>${hilla.version}</version>
+                <groupId>com.vaadin</groupId>
+                <artifactId>vaadin-bom</artifactId>
+                <version>${vaadin.version}</version>
                 <type>pom</type>
                 <scope>import</scope>
             </dependency>
@@ -34,12 +36,8 @@

     <dependencies>
         <dependency>
-            <groupId>dev.hilla</groupId>
-            <artifactId>hilla-react</artifactId>
-        </dependency>
-        <dependency>
-            <groupId>dev.hilla</groupId>
-            <artifactId>hilla-spring-boot-starter</artifactId>
+            <groupId>com.vaadin</groupId>
+            <artifactId>vaadin-spring-boot-starter</artifactId>
         </dependency>
         <dependency>
             <groupId>org.parttio</groupId>
@@ -84,9 +82,9 @@
             </plugin>

             <plugin>
-                <groupId>dev.hilla</groupId>
-                <artifactId>hilla-maven-plugin</artifactId>
-                <version>${hilla.version}</version>
+                <groupId>com.vaadin</groupId>
+                <artifactId>vaadin-maven-plugin</artifactId>
+                <version>${vaadin.version}</version>
                 <executions>
                     <execution>
                         <goals>
@@ -105,9 +103,9 @@
             <build>
                 <plugins>
                     <plugin>
-                        <groupId>dev.hilla</groupId>
-                        <artifactId>hilla-maven-plugin</artifactId>
-                        <version>${hilla.version}</version>
+                        <groupId>com.vaadin</groupId>
+                        <artifactId>vaadin-maven-plugin</artifactId>
+                        <version>${vaadin.version}</version>
                         <executions>
                             <execution>
                                 <goals>
```

### System Info

24.4.0.alpha7

setting manually version to 6.22.2 does not help:
```
$ grep router package.json
    "@vaadin/router": "1.7.5",
    "react-router-dom": "6.22.2"
    "@vaadin/hilla-file-router": "24.4.0-alpha7",
    "@vaadin/router": "$@vaadin/router",
    "react-router-dom": "$react-router-dom"
      "@vaadin/router": "1.7.5",
      "react-router-dom": "6.22.2"
      "@vaadin/hilla-file-router": "24.4.0-alpha7",
```

```
npm ERR! code ERESOLVE
npm ERR! ERESOLVE could not resolve
npm ERR!
npm ERR! While resolving: no-name@undefined
npm ERR! Found: react-router@6.20.1
npm ERR! node_modules/react-router
npm ERR!   react-router@"6.20.1" from react-router-dom@6.20.1
npm ERR!   node_modules/react-router-dom
npm ERR!     react-router-dom@"6.22.2" from the root project
npm ERR!
npm ERR! Could not resolve dependency:
npm ERR! dev @vaadin/hilla-file-router@"24.4.0-alpha7" from the root project
npm ERR!
npm ERR! Conflicting peer dependency: react-router@6.22.2
npm ERR! node_modules/react-router
npm ERR!   peer react-router@"^6.21.1" from @vaadin/hilla-file-router@24.4.0-alpha7
npm ERR!   node_modules/@vaadin/hilla-file-router
npm ERR!     dev @vaadin/hilla-file-router@"24.4.0-alpha7" from the root project
npm ERR!
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
```

--- a/packages/ts/hilla-file-router/package.json
+++ b/packages/ts/hilla-file-router/package.json
@@ -58,7 +58,6 @@
   },
   "peerDependencies": {
     "react": "^18.2.0",
-    "react-router": "^6.21.1",
     "react-router-dom": "^6.21.1"
   },
   "devDependencies": {

--- a/packages/ts/hilla-file-router/src/runtime/toReactRouter.ts
+++ b/packages/ts/hilla-file-router/src/runtime/toReactRouter.ts
@@ -1,5 +1,5 @@
 import { type ComponentType, createElement } from 'react';
-import type { RouteObject } from 'react-router';
+import type { RouteObject } from 'react-router-dom';
 import { convertComponentNameToTitle } from '../shared/convertComponentNameToTitle.js';
 import type { AgnosticRoute } from '../types.js';
 import { transformRoute } from './utils.js';

--- a/packages/ts/hilla-file-router/src/runtime/useViewConfig.ts
+++ b/packages/ts/hilla-file-router/src/runtime/useViewConfig.ts
@@ -1,5 +1,5 @@
 import type { UIMatch } from '@remix-run/router';
-import { useMatches } from 'react-router';
+import { useMatches } from 'react-router-dom';
 import type { ViewConfig } from '../types.js';
 
 /**

